assets/js/util/urls.js

```diff
@@ -85,3 +85,27 @@ export function normalizeURL( incomingURL ) {
 export function isHashOnly( url ) {
 	return /^#\w[A-Za-z0-9-_]*$/.test( url );
 }
+
+/**
+ * Shortens a URL to fit a given length.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} url      The original URL to shorten.
+ * @param {number} maxChars The maximum length of the URL.
+ * @return {string} The shortened URL.
+ */
+export function shortenURL( url, maxChars ) {
+	if ( url.length <= maxChars ) {
+		return url;
+	}
+
+	const urlObject = new URL( url );
+	const shortenedURL = url.replace( urlObject.origin, '' );
+	if ( shortenedURL.length < maxChars ) {
+		return shortenedURL;
+	}
+
+	const extraChars = shortenedURL.length - Math.floor( maxChars ) + 3; // 3 is the length of "...".
+	return '...' + shortenedURL.substr( extraChars );
```

Comment:
Arguably this should at least be an actual ellipsis:

```suggestion
	return '…' + shortenedURL.substr( extraChars );
```

But it's worth noting that this might not localise extremely well. Given it's a URL the RTL considerations probably don't matter too much, but what if there are locales where we don't trail off with "…"? That said, a quick web search shows this StackOverflow question that indicates a lot of locales _will_ use ellipsis, and WordPress itself seems to include it and it gets kept in the localisation: https://stackoverflow.com/questions/48718469/how-is-ellipsis-localized

Seems fine, but we should use the actual ellipsis character over three full-stops I think. 🙂